### ISSUE TITLE
[HOLD][Tracking] Add Native Share Menus to New Dot - Share Into New Expensify

### ISSUE THREAD

**Issue by grgia** [Expensify/App#15603]
## Design Doc
https://docs.google.com/document/d/10agvPMrVzXViEaH47gAXr9CZN62aaI3L5kj4aoj5LhI/edit

## Proposal

 Let’s implement native share menus so users can both share-to and share-from New Dot.

 **Problem:**
 There are plenty of times where I want to share something from outside our app into New Expensify. But there’s no way to quickly share links/photos/information while outside Expensify. For example, if I’m in the Photos App and I decide I want to share a photo, I’ll have to leave the app → open New Dot → go to the correct chat → click add attachments → find the same image again → and so on.
 On the flipside, there’s no easy way to export information from inside the app to outside Expensify. There’s no quick solution to export a PDF to print or to AirDrop a photo from one device to another.

 **Solution:**
 Let’s utilize one of our external agencies to implement Native Share Menus.
 Share menus are an essential feature to most other chat apps. If a user wants to send a photo, let’s make it as easy as possible. From the photo app you’d just click share → New Expensify and voila! In fact, we already have this feature for Old Dot. Right now, you can take a picture -> share -> Expensify, and immediately start a smartScan. So adding Share Menus to New Dot would be a requirement to maintain feature parity between Old and New Apps.
 As for sharing from inside Expensify to outside sources, to make New Expensify our go-to chat and finance collaboration app in the future, we should be able to communicate with those who have not yet adopted the app. For those of us that are eager to use NewDot as our primary chat app in our day-to-day lives, this feature would be a valuable way to encourage that while the rest of the world is still using WhatsApp. Later on, external sharing can feed into our viral business model, where chats/photos/links sent from New Expensify can be wrapped in an advertising moment, incentivizing people to join New Expensify from other chat apps.

https://expensify.slack.com/archives/CC7NECV4L/p1675375866420119

## Pre-designs
[Initial Pre-design](https://expensify.slack.com/archives/C01GTK53T8Q/p1678122006755739)
[Design Pre-design](https://expensify.slack.com/archives/C01GTK53T8Q/p1680526423839029)

## Tasks

- [x] Post **Proposal (full Problem/Solution statement)** in `#whatsnext`
- [x] Wait at least one full business day, and until the post has a majority (2/3) of positive reactions (👍)
- [x] Paste Proposal in the space above with a link to the Slack thread
- [x] Email `strategy@expensify.com` and paste in the Proposal
- [x] Fill out the **High-level overview of the problem**, **Timeline**, and **Terminology** sections of the Design Doc
- [x] Email `strategy@expensify.com` (continue the same email chain as before) with the link to your Design Doc
- [x] Host a **pre-design** meeting ([example](https://expensify.slack.com/archives/C01GTK53T8Q/p1665794669539419)) in an appropriate slack channel to discuss any necessary details in public before filling out the **High-level of proposed solution** section.
- [x] Fill out the **High-level of proposed solution** section
- [x] Email `stategy@expensify.com` again with links to the doc and pre-design conversation in Slack
- [x] Add the `DesignDocReview` label to get the **High-level of proposed solution** section reviewed
- [x] Respond to any questions or concerns and bring up blockers in Slack to get a consensus if necessary
- [ ] Confirm that the doc has the minimum necessary number of reviews before proceeding
- [ ] Host another pre-design meeting in the appropriate slack channel to ask for engineering feedback on the technical solution.
- [ ] Fill out the **Detailed implementation of the solution** and related sections.
- [ ] Re-add the `DesignDocReview` label to this issue
- [ ] Respond to any questions or concerns and bring up blockers in Slack to get consensus if necessary
- [ ] Confirm that the doc has the minimum necessary number of reviews before proceeding
- [ ] Email `strategy@expensify.com` one last time to let them know the Design Doc is moving into the implementation phase
- [ ] Implement the changes
- [ ] Send out a follow up email to `strategy@expensify.com` once everything has been implemented and do a **Project Wrap-Up** retrospective that provides:
    - Summary of what we accomplished with this project
    - What went well?
    - What could we have done better?
    - What did we learn?

## Related Issues

Mockup: https://github.com/Expensify/App/issues/16256

**Comment by MelvinBot** [comment]
Triggered auto assignment to @twisterdotcom (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by grgia** [comment]
Started a conversation about working with an external agency [here](https://expensify.slack.com/archives/C01SKUP7QR0/p1677610762419369)

**Comment by grgia** [comment]
Some related libraries:

React Native
- [Share](https://reactnative.dev/docs/share)

External (I suggest we consider these, as we will want the ability to better customize the share modal)
- [react-native-share](https://github.com/react-native-share/react-native-share) 
- [react-native-share-menu](https://github.com/meedan/react-native-share-menu)

**Comment by grgia** [comment]
![image](https://user-images.githubusercontent.com/38015950/228518728-ad5d84cd-6455-4afa-8151-98f6a44c21f6.png)

![image](https://user-images.githubusercontent.com/38015950/228518798-b941358f-6f5f-42e9-bbd7-08d4c8539e59.png)

**Comment by grgia** [comment]
@lindboe Could you comment on this so I can assign you?

**Comment by MelvinBot** [comment]
📣 @lindboe! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by roryabraham** [comment]
Super excited!

**Comment by MelvinBot** [comment]
:wave: Hello Generalist Track Team - you have been assigned to review this **High Level** Design Doc. Please, treat this as a priority. Review [this Stack Overflow](https://stackoverflow.com/c/expensify/questions/5838) for some tips on reviewing a design doc. Once you are done, simply press the `Add "Reviewed Doc" comment` button in the right hand side K2 pannel or follow [these](https://stackoverflow.com/c/expensify/questions/13713) instructions.

---
- `@arielgreen` (Expensifier / Graduate) - https://github.com/Expensify/App/issues/17334
- `@caitlinwhite1` (Expensifier / Graduate) - https://github.com/Expensify/App/issues/17335
- `@zsgreenwald` (Project Manager) - https://github.com/Expensify/App/issues/17336
- `@stitesExpensify` (Project Manager) - https://github.com/Expensify/App/issues/17337
- `@conorpendergrast` (Product Manager) - https://github.com/Expensify/App/issues/17338
- `@stephanieelliott` (Product Manager) - https://github.com/Expensify/App/issues/17339
- `@quinthar` (Generalist) - https://github.com/Expensify/App/issues/17340
- `@anmurali` (Generalist) - https://github.com/Expensify/App/issues/17341
- `@HaleyJacobson` (Accounting Technical Team) - https://github.com/Expensify/App/issues/17342

**Comment by stitesExpensify** [comment]
I have read and reviewed this Design Doc!

**Comment by roryabraham** [comment]
I have read and reviewed this Design Doc!

**Comment by stephanieelliott** [comment]
I have read and reviewed this Design Doc!

**Comment by Julesssss** [comment]
I have read and reviewed this Design Doc!

**Comment by conorpendergrast** [comment]
I have read and reviewed this Design Doc!

**Comment by caitlinwhite1** [comment]
I have read and reviewed this Design Doc!

**Comment by cjjohnson121** [comment]
I have read and reviewed this Design Doc!

**Comment by puneetlath** [comment]
I have read and reviewed this Design Doc!

**Comment by anmurali** [comment]
I have read and reviewed this Design Doc!

**Comment by arielgreen** [comment]
I have read and reviewed this Design Doc!

**Comment by jliexpensify** [comment]
I have read and reviewed this Design Doc!

**Comment by quinthar** [comment]
I have read and reviewed this Design Doc!

**Comment by grgia** [comment]
Not overdue, waiting on detailed section from @lindboe

**Comment by grgia** [comment]
in progress

**Comment by grgia** [comment]
Detailed Doc Ready For Review!

**Comment by Julesssss** [comment]
I have read and reviewed this Design Doc!

**Comment by puneetlath** [comment]
I have read and reviewed this Design Doc!

**Comment by grgia** [comment]
Detailed Doc Is In Review

**Comment by yuwenmemon** [comment]
I have read and reviewed this Design Doc!

**Comment by conorpendergrast** [comment]
I have read and reviewed this Design Doc!

**Comment by arielgreen** [comment]
I have read and reviewed this Design Doc!

**Comment by quinthar** [comment]
I have read and reviewed this Design Doc!

**Comment by quinthar** [comment]
I'm a broken record here, but I really, really hope we ditch the iOS share extension and just open the app.  It feels like 90% of the work is iOS specific, totally avoidable, and goes explicitly against our cross-platform philosophy.

**Comment by quinthar** [comment]
(But otherwise, it looks great and I can't wait to have it!!)

**Comment by anmurali** [comment]
I have read and reviewed this Design Doc!

**Comment by stephanieelliott** [comment]
I have read and reviewed this Design Doc!

**Comment by grgia** [comment]
Detailed is done, implementation was paused for wave projects. We are working on including scan pending this convo: 
https://expensify.slack.com/archives/C04KQL2HSQ1/p1696347905090599

**Comment by pecanoro** [comment]
I have read and reviewed this Design Doc!

**Comment by grgia** [comment]
Design Doc is being updated

**Comment by grgia** [comment]
@arielgreen @chiragsalian @zsgreenwald @stitesExpensify @robertjchen @pecanoro @puneetlath @yuwenmemon @conorpendergrast @stephanieelliott @anmurali @quinthar

We've updated the detailed doc to include smart scanning in V1. I re-added the label and it did not reassign you as reviewers, so if you're interested in taking a look at the update from the last review, please check out the following **updated DD** and leave a review! 

https://docs.google.com/document/d/10agvPMrVzXViEaH47gAXr9CZN62aaI3L5kj4aoj5LhI/edit

**Comment by johnmlee101** [comment]
I have read and reviewed this Design Doc!

**Comment by tgolen** [comment]
I have read and reviewed this Design Doc!

**Comment by puneetlath** [comment]
I have read and reviewed this Design Doc!

**Comment by chiragsalian** [comment]
I have read and reviewed this Design Doc!

**Comment by jliexpensify** [comment]
I have read and reviewed this Design Doc!

**Comment by trjExpensify** [comment]
I have read and reviewed this Design Doc!

**Comment by stitesExpensify** [comment]
I have read and reviewed this Design Doc!

**Comment by amyevans** [comment]
I have read and reviewed this Design Doc!

**Comment by stephanieelliott** [comment]
I have read and reviewed this Design Doc!

**Comment by conorpendergrast** [comment]
I have read and reviewed this Design Doc!

**Comment by arielgreen** [comment]
I have read and reviewed this Design Doc!

**Comment by grgia** [comment]
In implementation, was briefly put on hold for violations project work. This project has been added to wave5

**Comment by grgia** [comment]
Checking on the status of implementation with violations project

**Comment by lindboe** [comment]
@grgia Violations is still ongoing and at the current pace looks likely to continue into next week. I expect we'll be able to start picking up Share again mid-next-week.

**Comment by grgia** [comment]
Implementation is in progress

**Comment by JmillsExpensify** [comment]
I have read and reviewed this Design Doc!

**Comment by grgia** [comment]
Implementation is in progress, current ETA end of feb

**Comment by lindboe** [comment]
Hand-off information, to add on from the design doc:

For planned updates to react-native-share-menu, this is the root tracking issue: https://github.com/Expensify/react-native-share-menu/issues/294. There are linked issues that have some associated PRs already.

There's a google doc available here with extra notes and information from prior prototyping work here, some of which is for react-native-share-menu, and some is for New Expensify: https://docs.google.com/document/d/1ku-jZ9mjDkRgMHWGhe-bmuPAwK88sh5WgczU3vc2mfw/edit?usp=sharing.

**Comment by grgia** [comment]
Putting this project on HOLD temporarily while we discuss integrating with Hybrid Web

**Comment by trjExpensify** [comment]
👋 I haven't migrated this issue to the #wave-collect project. If this project doesn't get taken care of as part of HybridApp, bring it up in the #wave-collect channel. Thanks!
